generators/gae/index.js

```diff
@@ -0,0 +1,694 @@
+                this.gaeMinInstances = this.config.get('gaeMinInstances');
+            }
+        };
+    }
+
+    defaultProjectId() {
+        if (this.gcpProjectId) {
+            return this.gcpProjectId;
+        }
+        try {
+            const projectId = execSync('gcloud config get-value core/project --quiet', { encoding: 'utf8' });
+            return projectId.trim();
+        } catch (ex) {
+            this.log.error('Unable to determine the default Google Cloud Project ID');
+            return undefined;
+        }
+    }
+
+    defaultServiceNameChoices(defaultServiceExists) {
+        if (this.applicationType === 'monolith') {
+            return defaultServiceExists ? ['default', _.kebabCase(this.baseName)] : ['default'];
+        } else if (this.applicationType === 'gateway') {
+            return ['default'];
+        }
+
+        return [_.kebabCase(this.baseName)];
+    }
+
+    get prompting() {
+        return {
+            askForProjectId() {
+                if (this.abort) return;
+                const done = this.async();
+                const prompts = [
+                    {
+                        type: 'input',
+                        name: 'gcpProjectId',
+                        message: 'Google Cloud Project ID',
+                        default: this.defaultProjectId(),
+                        validate: (input) => {
+                            if (input.length === 0) {
+                                return 'Project ID cannot empty';
+                            }
+                            try {
+                                execSync(`gcloud projects describe ${input}`);
+                                this.gcpProjectIdExists = true;
+                            } catch (ex) {
+                                this.gcpProjectIdExists = false;
+                                return `Project ID "${chalk.cyan(input)}" does not exist, please create one first!`;
+                            }
+                            return true;
+                        }
+                    }];
+
+                this.prompt(prompts).then((props) => {
+                    this.gcpProjectId = props.gcpProjectId;
+                    done();
+                });
+            },
+
+            askForLocation() {
+                if (this.abort) return;
+                const done = this.async();
+
+                exec(`gcloud app describe --format="value(locationId)" --project="${this.gcpProjectId}"`, (err, stdout) => {
+                    if (err) {
+                        const prompts = [
+                            {
+                                type: 'list',
+                                name: 'gaeLocation',
+                                message: 'In which Google App Engine location do you want to deploy ?',
+                                choices: [
+                                    { value: 'northamerica-northeast1', name: 'northamerica-northeast1 - Montréal' },
```

Comment:
I think we can dynamically retrieve the list of regions and instance types from the cloud CLI.
Maybe this can be done in another PR, eg:
```
➜  ~ gcloud app regions list
REGION                   SUPPORTS STANDARD  SUPPORTS FLEXIBLE
asia-northeast1          YES                YES
asia-south1              YES                YES
australia-southeast1     YES                YES
europe-west              YES                YES
europe-west2             YES                YES
europe-west3             YES                YES
northamerica-northeast1  YES                YES
southamerica-east1       YES                YES
us-central               YES                YES
us-east1                 YES                YES
us-east4                 YES                YES
```

Reply:
Looks like we can get it as json with: `gcloud app regions list --format json`.